### PR TITLE
use compression level 1 by default, for speed

### PR DESCRIPTION
**Pull request by akiezun**
@droazen please review

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java [hunk 1]
```diff
@@ -142,6 +142,7 @@ public Object instanceMainPostParseArgs() {
                         " " + System.getProperty("java.runtime.version") +
                         "; Version: " + commandLineParser.getVersion() +
                         " " + (Defaults.USE_ASYNC_IO ? "asyncIO": "syncIO") +
+                        " " + "DefaultCompressionLevel:" + Defaults.COMPRESSION_LEVEL +
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: Formatting quibble: add a space after the colon

Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: Also, make it clear that this refers to BAM compression level only

File: README.md [hunk 2]
```diff
@@ -170,6 +170,12 @@ If you are looking for the codebase of the current production version of GATK, p
    JAVA_OPTS="-Dsamjdk.use_async_io=false" ./gatk-launch <rest of command>
 ```
 
+* By default, GATK (non-spark) uses compression level 1 for writting BAM files (fastest code but least compressed files). To change the default, run GATK like this:
+
+```
+   JAVA_OPTS="-Dsamjdk.compression_level=5" ./gatk-launch <rest of command>
```
Comment (droazen) [inline README.md]: How does `JAVA_OPTS interact with `applicationDefaultJvmArgs` in gradle? Does it supplement `applicationDefaultJvmArgs` or replace it completely?

**Comment by droazen** [comment]
What effect does this change have on the size of the bams we produce?

**Comment by droazen** [comment]
Review complete -- back to @akiezun. Merge after addressing comments and answering the question about the effect on bam size.

File: src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java (hunk 1 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: done

File: README.md (hunk 2 above)
Comment (akiezun) [inline README.md]: By using `jinfo -sysprops` i verified that using `JAVA_OPTS` supplements `applicationDefaultJvmArgs`, eg `JAVA_OPTS="-Dsamjdk.compression_level=5"` only changes the compression level and not other properties.

**Comment by akiezun** [comment]
I took a file of size 1265508369 bytes and run with JdkInflater at 10 levels.

**RESULT: Level 1 bams are 10% larger than level 5 but making them takes half the time. Definitely worth it IMO.**

level 0 3997540284 bytes  100%    0.62 minutes
level 1 1406624035 bytes   35%     1.14 minutes
level 2 1379001414 bytes   34%     1.22 minutes
level 3 1350188215 bytes   33%     1.40 minutes
level 4 1314300784 bytes   32%     1.62 minutes
level 5 1282932620 bytes   32%     2.03 minutes  <- 10% smaller than level 1
level 6 1265411149 bytes    31%     2.56 minutes
level 7 1258920244 bytes   31%     2.90 minutes
level 8 1252732792 bytes   31%     3.65 minutes
level 9 1251852390 bytes   31%     4.04 minutes

**Comment by akiezun** [comment]
@vdauwera see comment above - may be a fun fact for the gatk forum

**Comment by droazen** [comment]
Can you add `Level 1 bams are 10% larger than level 5 but making them takes half the time` to the relevant section of the README?

**Comment by akiezun** [comment]
@droazen done

File: README.md [hunk 3]
```diff
@@ -164,12 +164,18 @@ If you are looking for the codebase of the current production version of GATK, p
...
 ```
 
-* By default, GATK (non-spark) uses asynchronous IO and will use a separate thread for writting read and variant files. On systems with only 1 CPU this may reduce performance. To turn that option off, run GATK like this:
+* By default, GATK (non-spark) uses asynchronous IO and will use a separate thread for writing read and variant files. On systems with only 1 CPU this may reduce performance. To turn that option off, run GATK like this:
 
 ```
    JAVA_OPTS="-Dsamjdk.use_async_io=false" ./gatk-launch <rest of command>
 ```
 
+* By default, GATK (non-spark) uses compression level 1 for writing BAM files (fastest code but least compressed files). Level 1 BAM files are only 10% larger than level 5 but they take less half that time to create. To change the default, run GATK like this:
```
Comment (droazen) [inline README.md]: less -> less than

Comment (akiezun) [inline README.md]: fixed

**Pull request merged**